code/DDSCodeTester.cpp

```diff
@@ -3977,3 +3977,214 @@ void dds_persistence_examples()
     DataReader* reader = subscriber->create_datareader(topic, drqos);
     //!--
 }
+
+class LoanableHelloWorldPubSubType : public eprosima::fastdds::dds::TopicDataType
+{
+public:
+
+    LoanableHelloWorldPubSubType()
+        : TopicDataType()
+    {
+        setName("LoanableHelloWorld");
+    }
+
+    bool serialize(
+            void* /*data*/,
+            eprosima::fastrtps::rtps::SerializedPayload_t* /*payload*/) override
+    {
+        return true;
+    }
+
+    bool deserialize(
+            eprosima::fastrtps::rtps::SerializedPayload_t* /*payload*/,
+            void* /*data*/) override
+    {
+        return true;
+    }
+
+    std::function<uint32_t()> getSerializedSizeProvider(
+            void* /*data*/) override
+    {
+        return std::function<uint32_t()>();
+    }
+
+    void* createData() override
+    {
+        return nullptr;
+    }
+
+    void deleteData(
+            void* /*data*/) override
+    {
+    }
+
+    bool getKey(
+            void* /*data*/,
+            eprosima::fastrtps::rtps::InstanceHandle_t* /*ihandle*/,
+            bool /*force_md5*/) override
+    {
+        return true;
+    }
+
+    MD5 m_md5;
+    unsigned char* m_keyBuffer;
+};
+
+class LoanableHelloWorld
+{
+public:
+
+    LoanableHelloWorld()
+    {
+        m_index = 0;
+        memset(&m_message, 0, (256) * 1);
+    }
+
+    uint32_t& index()
+    {
+        return m_index;
+    }
+
+    uint32_t index() const
+    {
+        return m_index;
+    }
+
+    const std::array<char, 256>& message() const
+    {
+        return m_message;
+    }
+
+    std::array<char, 256>& message()
+    {
+        return m_message;
+    }
+
+private:
+
+    uint32_t m_index;
+    std::array<char, 256> m_message;
+};
+
+void dds_zero_copy_example()
+{
+    {
+        //LOANABLE_HELLOWORLD_EXAMPLE_WRITER
+        // CREATE THE PARTICIPANT
+        DomainParticipantQos pqos;
+        pqos.name("Participant_pub");
+        DomainParticipant* participant = DomainParticipantFactory::get_instance()->create_participant(0, pqos);
+        // REGISTER THE TYPE
+        TypeSupport type;
+        type.register_type(participant);
+        // CREATE THE PUBLISHER
+        Publisher* publisher = participant->create_publisher(PUBLISHER_QOS_DEFAULT, nullptr);
+        // CREATE THE TOPIC
+        Topic* topic = participant->create_topic(
+            "LoanableHelloWorldTopic",
+            type.get_type_name(),
+            TOPIC_QOS_DEFAULT);
+        // CREATE THE WRITER
+        DataWriterQos wqos = DATAWRITER_QOS_DEFAULT;
+        wqos.history().depth = 10;
+        wqos.durability().kind = TRANSIENT_LOCAL_DURABILITY_QOS;
+        // Enable the DataSharingQosPolicy using the DataWriterQoS
+        // wqos.data_sharing().automatic();
```

Comment:
Not needed, as it is `automatic` by default. May lead the user into thinking it is needed.
See my comment on the description of this code (section DataWriter)
Maybe changing as:
```suggestion
        // DataSharingQosPolicy has to be set to AUTO (the default) or ON
        // wqos.data_sharing().on("shared_directory");
```
and leaving it commented, like "this is also valid, but not required, and the default value is enough"

Reply:
:+1: